storage/writer.go

```diff
@@ -172,6 +172,25 @@ func (w *Writer) open() error {
 			// call to set up the upload as well as calls to upload individual chunks
 			// for a resumable upload (as long as the chunk size is non-zero). Hence
 			// there is no need to add retries here.
+
+			// Retry only when the operation is idempotent or the retry policy is RetryAlways.
+			var isIdempotent bool
+			if w.o.conds != nil && (w.o.conds.GenerationMatch >= 0 || w.o.conds.DoesNotExist == true) {
+				isIdempotent = true
+			}
```

Comment:
Here and below, isn't setting the variable to true redundant? 
```suggestion
				isIdempotent := w.o.conds != nil && (w.o.conds.GenerationMatch >= 0 || w.o.conds.DoesNotExist == true)
```

Reply:
Yep! I'll change this.